### PR TITLE
allow old epoch timestamps with 11-13 digits as epoch + milliseconds

### LINKED ISSUE #1295: Curator delete crashes if encountering an old timestamp
### Expected Behavior
Curator delete should not crash when trying to clean up indices

### Actual Behavior
Curator starts up and exits before any indices has been cleaned up with this error `Unusually formatted epoch timestamp. Should be 10, 13, or more digits`

We are running a curator job that runs this filter to look for old indices
```
- filtertype: age
      source: field_stats
      direction: older
      unit: days
      unit_count: 14
      field: '@timestamp'
      stats_result: max_value
```
However some host was started with an incorrect timestamp (year 1988), creating documents with a 12 digit timestamp, which was then set as the min_value for the index field_stats.

### Specifications

  - Version: ES 5.6.6, Curator 5.5.4

## Context (Environment)
We do index rollovers and deletions of indices older than 14 days. When a host sent in some docs with a very old timestamp from year 1988, Curator would crash complaining about a 12 digit timestamp, causing no indices to be deleted. With no old indices deleted we started growing in disk usage. Once the bad host was fixed, we tried to exclude the two indices we got with 12 digit timestamps but Curator runs the fix_epoch function on all indices at startup so exclude did not work.

## Detailed Description
I don't have the context for why 12 digit timestamps are not allowed in the first place, but I have created a code fix to allow it.

### PR DESCRIPTION
**Pull request by rsteneteg**
Fixes #1295 

## Proposed Changes
Classify all timestamp with 11 to 13 digits as epoch + milliseconds

### CODE REVIEW

**Comment by untergeek** [comment]
Have you rebased recently?  Multiple tests are failing, not just ones you added.

**Comment by rsteneteg** [comment]
I created the the fork of the repo a day or so ago, the last commit to master was 14 days ago so I expected my fork to be the same as your master. I just verified that I am in sync with your master.. so I am not sure why all those tests fail. Since those tests failed before I made any changes, I thought it was expected that those would fail, so I just made sure the test for the function I changed worked.

**Comment by untergeek** [comment]
All of the failed tests share this error message:

```
ValueError: Invalid epoch datatype received, expected an int
```

I'm going to go out on a limb and suggest that some other code needs changing.

**Comment by rsteneteg** [comment]
The fix_epoch function should work better now.. I was a bit to strict on the type check and did not think of strings that correctly converts to int. However I still get lots of other test failures that to me seems unrelated.

**Comment by rsteneteg** [comment]
ok.. Travis CI seems to have not had the same problems with the other tests that I had.. probably some environment issue for the integration tests.

**Comment by untergeek** [comment]
I will look this all over and merge it in a few days.  I'm super busy with work right now.

**Comment by rsteneteg** [comment]
Thanks you, we have mitigated the issue on our side so no need to rush, do it when you find time.

**Comment by untergeek** [comment]
Thanks for these changes

File: curator/utils.py [hunk 1]
```diff
@@ -235,19 +235,17 @@ def fix_epoch(epoch):
         even nanoseconds.
     :rtype: int
     """
+    if not isinstance(epoch, int):
+        raise ValueError('Invalid epoch datatype received, expected an int')
+
     # No decimals allowed
     epoch = int(epoch)
```
[the remark this answers is not in the document]
Comment (rsteneteg) [reply curator/utils.py]: Thanks, but those lines are outdated, replaced them with doing the `int(epoch)` within a try/catch block to capture bad input values.

**Pull request merged**